uber/models.py

```diff
@@ -1791,6 +1799,60 @@ def html(self):
             return SafeString(self.body.replace('\n', '<br/>'))
 
 
+class PromoCode(MagModel):
+    expiration_date = Column(UTCDateTime, default=c.ESCHATON)
+    price = Column(Integer, default=0)
+    code = Column(UnicodeText)
+    old_code = ""
+    # Uses should be -1 to make expired
+    uses = Column(Integer, nullable=True, default=None)
+    used_by = relationship(Attendee)
```

Comment:
The `backref` part in that relationship we established earlier (based on `group`) actually replaces this line, so you won't need it.